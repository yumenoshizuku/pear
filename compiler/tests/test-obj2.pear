def id,
Foo(bar) (
    works = 22 + 8,
    fun = 69 * 2,
    return bar + works,
)
Main() (
    tryme = Foo(),
    puts(tryme.fun),
    puts(tryme.works),
    puts(tryme(12)),
    puts(tryme(8)),
    tryme.works = 222,
    puts(tryme.works),
    tryme.fun = 21,
    puts(tryme.fun),
)

