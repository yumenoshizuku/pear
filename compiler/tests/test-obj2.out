#include <stdio.h>
#include <stdlib.h>
#include <gtk/gtk.h>
#include <glib.h>
#include <glib/gstdio.h>


int main(int argc, char** argv)
{
	
	gtk_init(&argc, &argv);
	printf("%d\n", 138);
	printf("%d\n", 30);
	printf("%d\n", 42);
	printf("%d\n", 38);
	printf("%d\n", 222);
	printf("%d\n", 21);
	gtk_main();
	return 0;
}

